src/pykeen/nn/utils.py

```diff
@@ -476,6 +502,45 @@ def get_image_paths(
         return [id_to_path.get(i) for i in ids]
 
 
+PYOBO_PREFIXES_WARNED = set()
+
+
+class PyOBOCache(TextCache):
+    """A cache that looks up labels of biomedical entities based on their CURIEs."""
+
+    def __init__(self, *args, **kwargs):
+        try:
+            import pyobo
+        except ImportError:
+            raise ImportError(f"Can not use {self.__class__.__name__} because pyobo is not installed.")
+        else:
+            self._get_name = pyobo.get_name
+        super().__init__(*args, **kwargs)
+
+    def get_texts(self, identifiers: Sequence[str]) -> Sequence[str]:
+        """Get text for the given CURIEs.
+
+        :param identifiers:
+            The compact URIs for each entity (e.g., ``['doid:1234', ...]``)
+
+        :return:
+            the label for each entity, looked up via :func:`pyobo.get_name`.
+        """
+        res = []
+        for curie in identifiers:
+            prefix, identifier = curie.split(":")
+            try:
+                name = self._get_name(prefix, identifier)
+            except subprocess.CalledProcessError:
+                if prefix not in PYOBO_PREFIXES_WARNED:
+                    logger.warning("could not get names from %s", prefix)
+                    PYOBO_PREFIXES_WARNED.add(prefix)
+                continue
```

Comment:
this is somewhat of a problem... what do we do for entries where we can't look up a label?

Reply:
Since the text representation is used as a way to initialize entity embeddings I think the way to go should be a non-text based initialization, which then gets replaced by text based representation. So here the TextCache's should should have some type of `null` value to indicate that no textual representation was present.

Alternatively an empty string could be used and the handling of these could be decided in the text encoders.

Reply:
Can we construct a somewhat meaningful string? e.g. something like "an entity of class X" or "an entity with prefix X"? At least for BERT-like models this could lead to somewhat useful representations.

Reply:
While I had a similar idea myself, afaik it remains to be tested whether this makes any significant difference and probably also depends on the downstream task. Since right know the entity name and the description is concatenated I would argue it should still produce pretty useful representations.

Reply:
I think we should change the interface to possibly return none then have the representation class deal with those, perhaps with one or more “policies” like you described